src/cargo.rs

```diff
@@ -73,8 +73,11 @@ impl Root {
 }
 
 /// Cargo project root
-pub fn root() -> Result<Option<Root>> {
-    let cd = env::current_dir().chain_err(|| "couldn't get current directory")?;
+pub fn root(manifest_path: Option<PathBuf>) -> Result<Option<Root>> {
+    let cd = match manifest_path {
+        Some(dir) => dir,
```

Comment:
`manifest_path` is not a directory but is already a path to a `Cargo.toml`.

Reply:
Thanks. I'd appreciate another look now.